scipy/stats/tests/test_stats.py

```diff
@@ -1242,6 +1242,220 @@ def test_kendalltau_nan_2nd_arg():
+                      + list(zip(alternatives, reversed(p_no_correlation),
+                                 [True]*3)))
+
+    @pytest.mark.parametrize("alternative, p_expected, rev", case_R_no_corr)
+    def test_against_R_no_correlation(self, alternative, p_expected, rev):
+        x, y = [1, 2, 3, 4, 5], [1, 5, 4, 2, 3]
+        stat_expected = 0
+        self.exact_test(x, y, alternative, rev, stat_expected, p_expected)
+
+    case_no_cor_b = (list(zip(alternatives, p_no_correlationb, [False]*3))
+                     + list(zip(alternatives, reversed(p_no_correlationb),
+                                [True]*3)))
+
+    @pytest.mark.parametrize("alternative, p_expected, rev", case_no_cor_b)
+    def test_against_R_no_correlationb(self, alternative, p_expected, rev):
+        x, y = [1, 2, 3, 4, 5, 6, 7, 8], [8, 6, 1, 3, 2, 5, 4, 7]
+        stat_expected = 0
+        self.exact_test(x, y, alternative, rev, stat_expected, p_expected)
+
+    case_R_lt_171 = (list(zip(alternatives, p_n_lt_171, [False]*3))
+                     + list(zip(alternatives, reversed(p_n_lt_171), [True]*3)))
+
+    @pytest.mark.parametrize("alternative, p_expected, rev", case_R_lt_171)
+    def test_against_R_lt_171(self, alternative, p_expected, rev):
+        # Data from Hollander & Wolfe (1973), p. 187f.
+        # Used from https://rdrr.io/r/stats/cor.test.html
+        x = [44.4, 45.9, 41.9, 53.3, 44.7, 44.1, 50.7, 45.2, 60.1]
+        y = [2.6, 3.1, 2.5, 5.0, 3.6, 4.0, 5.2, 2.8, 3.8]
+        stat_expected = 0.4444444444444445
+        self.exact_test(x, y, alternative, rev, stat_expected, p_expected)
+
+    case_R_lt_171b = (list(zip(alternatives, p_n_lt_171b, [False]*3))
+                      + list(zip(alternatives, reversed(p_n_lt_171b),
+                                 [True]*3)))
+
+    @pytest.mark.parametrize("alternative, p_expected, rev", case_R_lt_171b)
+    def test_against_R_lt_171b(self, alternative, p_expected, rev):
+        np.random.seed(0)
+        x = np.random.rand(100)
+        y = np.random.rand(100)
+        stat_expected = -0.04686868686868687
+        self.exact_test(x, y, alternative, rev, stat_expected, p_expected)
+
+    case_R_lt_171c = (list(zip(alternatives, p_n_lt_171c, [False]*3))
+                      + list(zip(alternatives, reversed(p_n_lt_171c),
+                                 [True]*3)))
+
+    @pytest.mark.parametrize("alternative, p_expected, rev", case_R_lt_171c)
+    def test_against_R_lt_171c(self, alternative, p_expected, rev):
+        np.random.seed(0)
+        x = np.random.rand(170)
+        y = np.random.rand(170)
+        stat_expected = 0.1115906717716673
+        self.exact_test(x, y, alternative, rev, stat_expected, p_expected)
+
+    case_R_gt_171 = (list(zip(alternatives, [False]*3)) +
+                     list(zip(alternatives, [True]*3)))
+
+    @pytest.mark.parametrize("alternative, rev", case_R_gt_171)
+    def test_against_R_gt_171(self, alternative, rev):
```

Comment:
How is this testing against R? I think you meant:

```suggestion
    case_gt_171 = (list(zip(alternatives, [False]*3)) +
                   list(zip(alternatives, [True]*3)))

    @pytest.mark.parametrize("alternative, rev", case_gt_171)
    def test_gt_171(self, alternative, rev):
```

Reply:
Maybe, also add tests against R for this case too just like you do with `n < 171`?

Reply:
You're right. This is named incorrectly. I'll commit this.

> Maybe, also add tests against R for this case too just like you do with n < 171?

I think the R implementation I was using refused to give an exact p-value for n > 171. I don't remember whether I looked hard for another implementation. 

However, as @Konrad0 suggested, I changed the cutoff for the switch between algorithms, and the tests still pass, so even though it's not part of continuous integration, we know that that the n > 171 code is also accurate.

Reply:
I tried testing against Matlab, which agrees closely with SciPy and R for the small samples tested (`lt_171`) but not exactly for large samples (`lt_171c`, `gt_171`). I wouldn't be able to tighten the tolerance compared to the existing test, which compared asymptotic vs exact, so I don't think I'll add that. I didn't easily find another R implementation that works for large samples, either, so I think I'll leave this alone.

Reply:
It's OK. I think this is already well tested. FWIW, no need to go too much out of way to add more!